### PR TITLE
ci: migration to GitHub Actions from CircleCI

### PR DESCRIPTION
**Pull request by uoboda-splunk**

### CODE REVIEW

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        uses: cycjimmy/semantic-release-action@v2.5.4
+        with:
+          semantic_version: 17
+          extra_plugins: |
+            @semantic-release/exec
+            @semantic-release/git
+            @google/semantic-release-replace-plugin
+        env:
+          GITHUB_TOKEN: ${{ secrets.SEMREL_TOKEN }}
+          PYPI_USERNAME: ${{ secrets.PYPI_USERNAME }}
```
Comment (uoboda-splunk) [inline .github/workflows/build-test-release.yml:262]: @arys-splunk I used semantic-release implementation from `addonfactory-ucc-generator` do you have access to `PYPI` creds? If yes can you also set them for this repo?

Comment (arys-splunk) [reply .github/workflows/build-test-release.yml:262]: nope, I do not. Ping Ryan please.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (arys-splunk) [inline .github/dependabot.yml:1]: Do we need lines 6-9 here?

File: .github/workflows/agreements.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: "CLA Assistant"
```
Comment (arys-splunk) [inline .github/workflows/agreements.yaml:1]: Can you please clean up comments?

File: .github/workflows/build-test-release.yml [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Install actionlint
+        run: |
+          bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/v1.6.3/scripts/download-actionlint.bash)
```
Comment (arys-splunk) [inline .github/workflows/build-test-release.yml:62]: If we are using this one, can you please add `actionlint` to `.pre-commit-config.yaml` file as well?
Example: https://github.com/splunk/addonfactory-ucc-generator/blob/3176ac155501ceab263a1f7a64a02d6a34aa31b4/.pre-commit-config.yaml#L46-L54

File: .releaserc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    ],
+  plugins:
+    [
+      "@semantic-release/commit-analyzer",
+      [
+        "@google/semantic-release-replace-plugin",
+        {
+          "replacements": [
+            {
+              "files": ["splunktaucclib/__init__.py"],
```
Comment (arys-splunk) [inline .releaserc:31]: This one should be
```suggestion
              "files": ["solnlib/__init__.py"],
```

File: .releaserc [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        {
+          "prepareCmd": "./.github/workflows/sr-prepare.sh ${nextRelease.version}",
+          "publishCmd": "./.github/workflows/sr-release.sh",
+          "shell": "bash"
+        },
+      ],
+      [
+        "@semantic-release/git",
+        {
+          "assets": ["NOTICE", "pyproject.toml", "splunktaucclib/__init__.py"],
```
Comment (arys-splunk) [inline .releaserc:73]:
```suggestion
          "assets": ["NOTICE", "pyproject.toml", "solnlin/__init__.py"],
```

File: .releaserc [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        "@google/semantic-release-replace-plugin",
+        {
+          "replacements": [
+            {
+              "files": ["splunktaucclib/__init__.py"],
+              "from": "__version__ ?=.*",
+              "to": "__version__ = \"${nextRelease.version}\"",
+              "results": [
+                {
+                  "file": "splunktaucclib/__init__.py",
```
Comment (arys-splunk) [inline .releaserc:36]:
```suggestion
                  "file": "solnlib/__init__.py",
```

**Review by arys-splunk — CHANGES_REQUESTED** [review]

File: .github/workflows/agreements.yaml (hunk 3 above)
Comment (uoboda-splunk) [reply .github/workflows/agreements.yaml:1]: done

**Comment by uoboda-splunk** [comment]
> Please use conventional commit messages: https://www.conventionalcommits.org/en/v1.0.0/

Changed PR name, I will use squash and merge option

File: .github/workflows/build-test-release.yml [hunk 8]
```diff
@@ -5,7 +5,7 @@ on:
     branches:
       - "main"
       - "develop"
-      - "github-actions-migration"
+      - "pre-commit"
```
Comment (arys-splunk) [inline .github/workflows/build-test-release.yml:8]:
```suggestion
      - "github-actions-migration"
```

**Pull request merged**